src/nvim/api/command.c

```diff
@@ -706,8 +696,8 @@ static bool string_iswhite(String str)
 /// Build cmdline string for command, used by `nvim_cmd()`.
 ///
 /// @return OK or FAIL.
-static void build_cmdline_str(char **cmdlinep, exarg_T *eap, CmdParseInfo *cmdinfo, char **args,
-                              size_t argc)
+static void build_cmdline_str(char **cmdlinep, exarg_T *eap, CmdParseInfo *cmdinfo,
+                              ArrayOf(String) args, size_t argc)
 {
   StringBuilder cmdline = KV_INITIAL_VALUE;
```

Comment:
Have you tried this?
```suggestion
  StringBuilder cmdline = KV_INITIAL_VALUE;
  kv_resize(cmdline, 1024);
```

Reply:
As per bfredl's suggestion, I've gone with `32` for now.